git-branch-linearity.sh

```diff
@@ -1,5 +1,22 @@
 #!/bin/sh
-out=$(git log origin/master..HEAD --merges --oneline)
+
+TARGET_BRANCH="master"
+
+for arg in "$@"
+do
+  case $arg in
+    -t=*|--target-branch=*)
+    TARGET_BRANCH="${arg#*=}"
+    shift # past argument name
+    ;;
+    *)    # unknown option
+	  echo "Unknow option '$1'"
+    exit 1
+    ;;
+  esac
+done
```

Comment:
Alternatively we could guess the branch name:

```bash
if [ $(git branch --list main) ]; then
    TARGET_BRANCH="main"
else
    TARGET_BRANCH="master"
fi

echo "Target branch $TARGET_BRANCH"

out=$(git log origin/${TARGET_BRANCH}..HEAD --merges --oneline)
```

Reply:
Side note: We might want to pass `${{ github.base.ref }}` (there is no circleCI equivalent).
This would be useful for PR targetting a release branch.

Reply:
I have take the auto detection 👍🏻